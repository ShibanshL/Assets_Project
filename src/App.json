{"next":"https://api.dev.parham.in/api/org/18/asset/?page=2&page_size=20","previous":null,"count":100,"current":1,"pages":5,"results":[{"host":"192.168.00.16","display_name":"device_00.16","unique_id":"0415cafd-74a4-4662-8e52-5e48bf1a5a56","tags":["database 16","server 16"],"type":"infra","is_live":null,"time_to_decommission":null,"created_at":"2022-07-20T09:10:53.432472Z","updated_at":"2022-07-20T09:10:53.432489Z","decommission_date":null,"org_id":18,"latest_risk_score":null,"scan_cycle_count":0,"vuln_count":0,"vuln_breakup":{},"last_scan":{},"cve_status_distribution":{}},{"host":"192.168.00.74","display_name":"device_00.74","unique_id":"046fdd69-975a-457e-ab11-7b086fdda3ee","tags":["database 74","server 74"],"type":"infra","is_live":null,"time_to_decommission":null,"created_at":"2022-07-20T09:10:56.236104Z","updated_at":"2022-07-20T09:10:56.236123Z","decommission_date":null,"org_id":18,"latest_risk_score":null,"scan_cycle_count":0,"vuln_count":0,"vuln_breakup":{},"last_scan":{},"cve_status_distribution":{}},{"host":"192.168.01.00","display_name":"device_01.00","unique_id":"07950c6f-9de5-489a-80b5-d1e16cd8d45f","tags":["database 100","server 100"],"type":"infra","is_live":null,"time_to_decommission":null,"created_at":"2022-07-20T09:10:57.438129Z","updated_at":"2022-07-20T09:10:57.438147Z","decommission_date":null,"org_id":18,"latest_risk_score":null,"scan_cycle_count":0,"vuln_count":0,"vuln_breakup":{},"last_scan":{},"cve_status_distribution":{}},{"host":"192.168.00.71","display_name":"device_00.71","unique_id":"0849f852-6811-4119-9880-a89a6ce5ef46","tags":["database 71","server 71"],"type":"infra","is_live":null,"time_to_decommission":null,"created_at":"2022-07-20T09:10:56.066992Z","updated_at":"2022-07-20T09:10:56.067010Z","decommission_date":null,"org_id":18,"latest_risk_score":null,"scan_cycle_count":0,"vuln_count":0,"vuln_breakup":{},"last_scan":{},"cve_status_distribution":{}},{"host":"192.168.00.73","display_name":"device_00.73","unique_id":"0ea8080c-f8ec-4a30-9dff-72b2d705e9b4","tags":["database 73","server 73"],"type":"infra","is_live":null,"time_to_decommission":null,"created_at":"2022-07-20T09:10:56.166251Z","updated_at":"2022-07-20T09:10:56.166269Z","decommission_date":null,"org_id":18,"latest_risk_score":null,"scan_cycle_count":0,"vuln_count":0,"vuln_breakup":{},"last_scan":{},"cve_status_distribution":{}},{"host":"192.168.00.96","display_name":"device_00.96","unique_id":"0ebc2264-7be5-4884-9bde-4637ede36bdc","tags":["database 96","server 96"],"type":"infra","is_live":null,"time_to_decommission":null,"created_at":"2022-07-20T09:10:57.251392Z","updated_at":"2022-07-20T09:10:57.251410Z","decommission_date":null,"org_id":18,"latest_risk_score":null,"scan_cycle_count":0,"vuln_count":0,"vuln_breakup":{},"last_scan":{},"cve_status_distribution":{}},{"host":"192.168.00.38","display_name":"device_00.38","unique_id":"0ee97fd7-c360-4514-b69e-00ca33582fab","tags":["database 38","server 38"],"type":"infra","is_live":null,"time_to_decommission":null,"created_at":"2022-07-20T09:10:54.581739Z","updated_at":"2022-07-20T09:10:54.581757Z","decommission_date":null,"org_id":18,"latest_risk_score":null,"scan_cycle_count":0,"vuln_count":0,"vuln_breakup":{},"last_scan":{},"cve_status_distribution":{}},{"host":"192.168.00.15","display_name":"device_00.15","unique_id":"1356b69b-7564-42ed-ad48-4890f20ebdfb","tags":["database 15","server 15"],"type":"infra","is_live":null,"time_to_decommission":null,"created_at":"2022-07-20T09:10:53.364057Z","updated_at":"2022-07-20T09:10:53.364075Z","decommission_date":null,"org_id":18,"latest_risk_score":null,"scan_cycle_count":0,"vuln_count":0,"vuln_breakup":{},"last_scan":{},"cve_status_distribution":{}},{"host":"192.168.00.42","display_name":"device_00.42","unique_id":"1591590a-65c2-4871-bd0c-1717d40ff3e1","tags":["database 42","server 42"],"type":"infra","is_live":null,"time_to_decommission":null,"created_at":"2022-07-20T09:10:54.767574Z","updated_at":"2022-07-20T09:10:54.767591Z","decommission_date":null,"org_id":18,"latest_risk_score":null,"scan_cycle_count":0,"vuln_count":0,"vuln_breakup":{},"last_scan":{},"cve_status_distribution":{}},{"host":"192.168.00.37","display_name":"device_00.37","unique_id":"1a92c7c3-1462-439c-bce4-a86634161066","tags":["database 37","server 37"],"type":"infra","is_live":null,"time_to_decommission":null,"created_at":"2022-07-20T09:10:54.555226Z","updated_at":"2022-07-20T09:10:54.555244Z","decommission_date":null,"org_id":18,"latest_risk_score":null,"scan_cycle_count":0,"vuln_count":0,"vuln_breakup":{},"last_scan":{},"cve_status_distribution":{}},{"host":"192.168.00.20","display_name":"device_00.20","unique_id":"1d64fc20-1ea1-4746-b146-32ef923439ce","tags":["database 20","server 20"],"type":"infra","is_live":null,"time_to_decommission":null,"created_at":"2022-07-20T09:10:53.565680Z","updated_at":"2022-07-20T09:10:53.565698Z","decommission_date":null,"org_id":18,"latest_risk_score":null,"scan_cycle_count":0,"vuln_count":0,"vuln_breakup":{},"last_scan":{},"cve_status_distribution":{}},{"host":"192.168.00.77","display_name":"device_00.77","unique_id":"20918885-281c-4ba7-8d7f-f31005d3bc08","tags":["database 77","server 77"],"type":"infra","is_live":null,"time_to_decommission":null,"created_at":"2022-07-20T09:10:56.358693Z","updated_at":"2022-07-20T09:10:56.358711Z","decommission_date":null,"org_id":18,"latest_risk_score":null,"scan_cycle_count":0,"vuln_count":0,"vuln_breakup":{},"last_scan":{},"cve_status_distribution":{}},{"host":"192.168.00.88","display_name":"device_00.88","unique_id":"2258839c-2635-4f36-95f7-9bb55a59bc24","tags":["database 88","server 88"],"type":"infra","is_live":null,"time_to_decommission":null,"created_at":"2022-07-20T09:10:56.854550Z","updated_at":"2022-07-20T09:10:56.854640Z","decommission_date":null,"org_id":18,"latest_risk_score":null,"scan_cycle_count":0,"vuln_count":0,"vuln_breakup":{},"last_scan":{},"cve_status_distribution":{}},{"host":"192.168.00.67","display_name":"device_00.67","unique_id":"262a9644-a56d-466d-bb1b-ca2ef7db02aa","tags":["database 67","server 67"],"type":"infra","is_live":null,"time_to_decommission":null,"created_at":"2022-07-20T09:10:55.931853Z","updated_at":"2022-07-20T09:10:55.931870Z","decommission_date":null,"org_id":18,"latest_risk_score":null,"scan_cycle_count":0,"vuln_count":0,"vuln_breakup":{},"last_scan":{},"cve_status_distribution":{}},{"host":"192.168.00.25","display_name":"device_00.25","unique_id":"29163e1a-629b-487b-91ac-504a7082b0a5","tags":["database 25","server 25"],"type":"infra","is_live":null,"time_to_decommission":null,"created_at":"2022-07-20T09:10:53.832944Z","updated_at":"2022-07-20T09:10:53.832963Z","decommission_date":null,"org_id":18,"latest_risk_score":null,"scan_cycle_count":0,"vuln_count":0,"vuln_breakup":{},"last_scan":{},"cve_status_distribution":{}},{"host":"192.168.00.34","display_name":"device_00.34","unique_id":"29d11ed3-4e4f-423e-96e2-e2bce5e309bd","tags":["database 34","server 34"],"type":"infra","is_live":null,"time_to_decommission":null,"created_at":"2022-07-20T09:10:54.451193Z","updated_at":"2022-07-20T09:10:54.451210Z","decommission_date":null,"org_id":18,"latest_risk_score":null,"scan_cycle_count":0,"vuln_count":0,"vuln_breakup":{},"last_scan":{},"cve_status_distribution":{}},{"host":"192.168.00.69","display_name":"device_00.69","unique_id":"2b46d052-b228-4bc1-94ac-e0b490ed727a","tags":["database 69","server 69"],"type":"infra","is_live":null,"time_to_decommission":null,"created_at":"2022-07-20T09:10:55.970235Z","updated_at":"2022-07-20T09:10:55.970253Z","decommission_date":null,"org_id":18,"latest_risk_score":null,"scan_cycle_count":0,"vuln_count":0,"vuln_breakup":{},"last_scan":{},"cve_status_distribution":{}},{"host":"192.168.00.48","display_name":"device_00.48","unique_id":"2dc3ed4f-73df-425c-b0a6-7699eac1bd2c","tags":["database 48","server 48"],"type":"infra","is_live":null,"time_to_decommission":null,"created_at":"2022-07-20T09:10:55.058845Z","updated_at":"2022-07-20T09:10:55.058864Z","decommission_date":null,"org_id":18,"latest_risk_score":null,"scan_cycle_count":0,"vuln_count":0,"vuln_breakup":{},"last_scan":{},"cve_status_distribution":{}},{"host":"192.168.00.98","display_name":"device_00.98","unique_id":"31913ba7-09a5-4bab-baa5-ddb4fd614577","tags":["database 98","server 98"],"type":"infra","is_live":null,"time_to_decommission":null,"created_at":"2022-07-20T09:10:57.345821Z","updated_at":"2022-07-20T09:10:57.345838Z","decommission_date":null,"org_id":18,"latest_risk_score":null,"scan_cycle_count":0,"vuln_count":0,"vuln_breakup":{},"last_scan":{},"cve_status_distribution":{}},{"host":"192.168.00.75","display_name":"device_00.75","unique_id":"31cbd29f-59df-4123-9fbe-4a32748294e9","tags":["database 75","server 75"],"type":"infra","is_live":null,"time_to_decommission":null,"created_at":"2022-07-20T09:10:56.255987Z","updated_at":"2022-07-20T09:10:56.256007Z","decommission_date":null,"org_id":18,"latest_risk_score":null,"scan_cycle_count":0,"vuln_count":0,"vuln_breakup":{},"last_scan":{},"cve_status_distribution":{}}],"page_size":20,"page_size_query_param":"page_size"}