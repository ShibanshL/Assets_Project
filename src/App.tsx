import { useState } from 'react'
import { AppShell, Center, Container,Group,Grid, Navbar} from '@mantine/core';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'; 
import {QueryClientProvider, QueryClient} from 'react-query'
import Test_File_1 from './Test_Files/Test_File_1';
import Test_File_2 from './Test_Files/Test_File_2';

const reactQuery = new QueryClient()

function App() {
  return (
    <QueryClientProvider client={reactQuery}>
      <Router>
        <AppShell
          padding={0}
          styles={(theme) => ({
            main: {background:'white',backgroundSize:'cover'},
          })}>
            <Container size={1920} p='0' m='0' style={{}} fluid>
                <Grid>
                  <Grid.Col span={12}><Test_File_1/></Grid.Col>
                  <Grid.Col span={12}><Test_File_2/></Grid.Col>
                </Grid>
            </Container>
        </AppShell>
      </Router>
    </QueryClientProvider>
  )
}

export default App
