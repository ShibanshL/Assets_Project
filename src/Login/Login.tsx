import React from 'react'
import { Container, Group, Button, TextInput, Text} from '@mantine/core';
import { useNavigate } from 'react-router-dom';
import { useForm } from '@mantine/form';
import {useMutation} from 'react-query' 
import {BsEyeFill,BsEyeSlashFill} from 'react-icons/bs'
import axios from 'axios';
import { useStore_1, useStore_2, useStore_3} from '../Store';
import { constants } from 'buffer';


var k = 0

var invalid;

interface api {
  data:{
    auth_token: string
  }
}

function Login() {

  const [pass,setPass] = React.useState('password')
  const setLog1 = useStore_2(state => state.setLog_1)
  const logData = useStore_3(state => state.logData)
  const setLogData = useStore_3(state => state.setLogData)
  const setToken = useStore_1(state => state.setToken)
  const Token = useStore_1(state => state.token)
  const [invalidCred,setInvalidCred] = React.useState('')

  let nav = useNavigate()

  //Mantine's basic useform that handles all the input and input authorization
  const form = useForm({
      initialValues: {
          email: '',
          password:'',
    },
    validate: {
        email: (value) => (/^\S+@\S+$/.test(value) ? null : 'Invalid email'),
        password : (value) => (value.length < 4)? 'Please Enter a biggger Password' : null
        },
    });

    const Password = () => {
        if(k%2 == 0){
          setPass('text')
          console.log('Icon = ',pass)
          k++
        }
        else{
          setPass('password')
          console.log('Icon = ',pass)
          k++
        }
      }
      
    //Here i am using react querry to send the post request
    const mutation = useMutation((values:any) => handleSubmit(values),
    {
      onSuccess: () => {
        console.log('W O R K I N G');
        console.log('DATA : ')
      }
      })

      //This is the function where the post request happen, here after post request i receive an authkey and i use localStorage
      //to keep the value of that auth key
      const handleSubmit = async (values:any) => {
        values.preventDefault()
        var em = form.values.email
        var pa = form.values.password

        console.log('Email : ',em+' Password : ',pa)

        let payload = {email:em,password:pa}

        axios.post(`${import.meta.env.VITE_URL}/api/auth/token/login/`,payload)
        .then((res) => {
              setToken(res?.data?.auth_token);
              if(res?.data?.auth_token){
                nav('/Assets')
              }
              else return 
            })
        .catch(err => {
          console.log(err)
          setInvalidCred('Invalid credentials, please Reload and Try again.')
        })

        setLogData(true)

        setLog1()

       
      }
    
    return (
      <Container pt='10px' pb='65vh' style={{background:'#f8f9fa'}} fluid>
        <Group sx={{maxWidth:400}} mx="auto" p='5px' style={{background:'white',borderRadius:'10px'}} direction='column' align='top' position='center'>
            <form onSubmit={(values:any) => handleSubmit(values)} style={{width:'100%'}}>
              <TextInput
              required
              label="Email"
              placeholder="Enter your Email"
              radius="sm"
              p='10px'
              {...form.getInputProps('email')}
              
              />
              <TextInput
              required
              type={pass}
              label="Password"
              placeholder="Password"
              radius="sm"
              pt='5px'
              p='10px'
              rightSection ={k%2 == 0?<BsEyeSlashFill style={{cursor:'pointer'}} onClick={Password}/>:<BsEyeFill style={{cursor:'pointer'}} onClick={Password}/>}
              {...form.getInputProps(pass)}
              
              />
              <Text color='red' ml='10px' pb='5px'>{invalidCred}</Text>
              {/* <Group position="right" mt="md"> */}
              <Button radius={'sm'} ml='10px' mt='5px' type="submit">Login</Button>
              {/* </Group> */}
            </form>
        </Group>
      </Container>
    )
}

export default Login