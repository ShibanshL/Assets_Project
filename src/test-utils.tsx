/* eslint-disable import/export */
import { cleanup, render } from "@testing-library/react";
import { afterEach } from "vitest";
import { QueryClient, QueryClientProvider } from "react-query";
import { BrowserRouter } from "react-router-dom";
import { rest } from "msw";
import axios from "axios";
import { PostLoginResMock } from "./getLoginResMock";

afterEach(() => {
  cleanup();
});
axios.defaults.headers.common["Authorization"] =
  "Token 8431916c2a4e2f97718632fc1a912c363a80bac9";
export const handlers = [
  rest.post("*/api/auth/token/login/", (req, res, ctx) => {
    // req.headers;
    return res(
      ctx.status(200),
      ctx.set({
        Authorization: `Token hkshfkhfhdsifhdikufsdki`,
        "Content-Type": ["application/json", "application/hal+json"],
        "Content-Length": "40032",
      }),
      ctx.json(PostLoginResMock)
    );
  }),
];
const reactQuery = new QueryClient();

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, {
    // wrap provider(s) here if needed
    wrapper: ({ children }) => children,
    ...options,
  });

const createTestQueryClient = () =>
  new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
    // logger: {
    //     log: console.log,
    //     warn: console.warn,
    //     error: () => {},
    // }
  });

export function renderWithClient(ui: React.ReactElement) {
  const testQueryClient = createTestQueryClient();
  const { rerender, ...result } = render(
    <QueryClientProvider client={testQueryClient}>{ui}</QueryClientProvider>
  );
  return {
    ...result,
    rerender: (rerenderUi: React.ReactElement) =>
      rerender(
        <QueryClientProvider client={testQueryClient}>
          {rerenderUi}
        </QueryClientProvider>
      ),
  };
}

export function createWrapper() {
  const testQueryClient = createTestQueryClient();
  return ({ children }: { children: React.ReactNode }) => (
    <QueryClientProvider client={testQueryClient}>
      {children}
    </QueryClientProvider>
  );
}

export function newRenderClient(Name: any) {
  return (
    <QueryClientProvider client={reactQuery}>
      <BrowserRouter>{Name}</BrowserRouter>
    </QueryClientProvider>
  );
}

export * from "@testing-library/react";
export { default as userEvent } from "@testing-library/user-event";
// override render export
export { renderWithClient as render };
