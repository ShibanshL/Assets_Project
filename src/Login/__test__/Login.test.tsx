import { describe, expect, it } from 'vitest'
import App from '../../App'
import Login from '../Login'
import Test_File_1 from '../../Test_Files/Test_File_1'
import { render, screen, userEvent } from '../../test-utils'
import { HiLogin } from 'react-icons/hi'
import { BrowserRouter } from 'react-router-dom'
import {QueryClientProvider, QueryClient } from 'react-query'
import { renderWithClient } from '../../test-utils'
import { newRenderClient } from '../../test-utils'

const reactQuery = new QueryClient()
// const Mock = (history:any) => {
//     return(
//         <QueryClientProvider client={reactQuery}>
//             <BrowserRouter>
//                 <Login />
//             </BrowserRouter>
//         </QueryClientProvider>
//     )
// }

describe('Login', () => {

    it('check someting', async() => {
        // const res = renderWithClient(<Login />)
        render(newRenderClient(<Login />))
        const name = screen.getByText('Email')
        expect(name).toBeInTheDocument()
    })

    it('check someting_1', () => {
        render(newRenderClient(<Login />))
        const name = screen.getByText('Email')
        expect(name).toBeTruthy()
    })

    it('check someting_2', () => {
        render(newRenderClient(<Login />))
        const name = screen.getByText('Email')
        expect(name).toContainHTML('label')
    })

    it('check someting_3', () => {
        render(newRenderClient(<Login />))
        const name = screen.getByPlaceholderText('Enter your Email')
        expect(name).toContainHTML('input')
    })

    it('check someting_4', () => {
        render(newRenderClient(<Login />))
        const name = screen.getByPlaceholderText('Enter your Email')
        expect(name).not.toContainHTML('button')
    })

    
    // it('check someting_5', async () => {
    //     render(newRenderClient(<Login />))
    //     const name = screen.getByPlaceholderText('Enter your Email')
    //     const pass = screen.getByPlaceholderText('Password')
    //     const button_Log = screen.getByRole('button',{name:'Login'})
    //     userEvent.type(name, 'shibansh@parham.in')
    //     userEvent.type(pass,'123456789')
    //     userEvent.click(button_Log)
    //     const url = await window.location.pathname
    //     expect(url).toBe('/Assets')
    //     // expect(name).not.toContainHTML('button')
    // })

    it('check someting_6', async () => {
        render(newRenderClient(<Login />))
        const name = screen.getByPlaceholderText('Enter your Email')
        const pass = screen.getByPlaceholderText('Password')
        const button_Log = screen.getByRole('button',{name:'Login'})
        userEvent.type(name, 'shibansh@parham.in')
        userEvent.type(pass,'123456897')
        userEvent.click(button_Log)
        const invalid = await console.warn('AxiosError')
        expect(invalid).toBeVisible()
        // expect(name).toBeInTheDocument()
    })


}) 