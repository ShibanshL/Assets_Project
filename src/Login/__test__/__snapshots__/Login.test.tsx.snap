// Vitest Snapshot v1

exports[`check someting 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mantine-Container-root mantine-1jly7oz"
        style="background: rgb(248, 249, 250);"
      >
        <div
          class="mantine-Group-root mantine-x03taj"
          style="background: white; border-radius: 10px;"
        >
          <form
            class="mantine-gnzaph mantine-Group-child"
            style="width: 100%;"
          >
            <div
              class="mantine-TextInput-root mantine-10rtrbx"
            >
              <label
                class="mantine-1w07r5r mantine-TextInput-label"
                for="mantine-exqmqpxs1"
                id="mantine-exqmqpxs1-label"
              >
                Email
                <span
                  aria-hidden="true"
                  class="mantine-1m203yh mantine-TextInput-required"
                >
                   *
                </span>
              </label>
              <div
                class="mantine-TextInput-wrapper mantine-12sbrde"
              >
                <input
                  aria-invalid="false"
                  class="mantine-TextInput-defaultVariant mantine-TextInput-input mantine-g773hu"
                  id="mantine-exqmqpxs1"
                  placeholder="Enter your Email"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="mantine-TextInput-root mantine-1x9761l"
            >
              <label
                class="mantine-1w07r5r mantine-TextInput-label"
                for="mantine-zwd3hgd2t"
                id="mantine-zwd3hgd2t-label"
              >
                Password
                <span
                  aria-hidden="true"
                  class="mantine-1m203yh mantine-TextInput-required"
                >
                   *
                </span>
              </label>
              <div
                class="mantine-TextInput-wrapper mantine-12sbrde"
              >
                <input
                  aria-invalid="false"
                  class="mantine-TextInput-defaultVariant mantine-TextInput-input mantine-15cu3qq"
                  id="mantine-zwd3hgd2t"
                  placeholder="Password"
                  required=""
                  type="password"
                  value=""
                />
                <div
                  class="mantine-1n9cu5e mantine-TextInput-rightSection"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    style="cursor: pointer;"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m10.79 12.912-1.614-1.615a3.5 3.5 0 0 1-4.474-4.474l-2.06-2.06C.938 6.278 0 8 0 8s3 5.5 8 5.5a7.029 7.029 0 0 0 2.79-.588zM5.21 3.088A7.028 7.028 0 0 1 8 2.5c5 0 8 5.5 8 5.5s-.939 1.721-2.641 3.238l-2.062-2.062a3.5 3.5 0 0 0-4.474-4.474L5.21 3.089z"
                    />
                    <path
                      d="M5.525 7.646a2.5 2.5 0 0 0 2.829 2.829l-2.83-2.829zm4.95.708-2.829-2.83a2.5 2.5 0 0 1 2.829 2.829zm3.171 6-12-12 .708-.708 12 12-.708.708z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="mantine-Text-root mantine-y2od05"
            />
            <button
              class="mantine-Button-filled mantine-Button-root mantine-1fhoz4r"
              type="submit"
            >
              <div
                class="mantine-3xbgk5 mantine-Button-inner"
              >
                <span
                  class="mantine-qo1k2 mantine-Button-label"
                >
                  Login
                </span>
              </div>
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mantine-Container-root mantine-1jly7oz"
      style="background: rgb(248, 249, 250);"
    >
      <div
        class="mantine-Group-root mantine-x03taj"
        style="background: white; border-radius: 10px;"
      >
        <form
          class="mantine-gnzaph mantine-Group-child"
          style="width: 100%;"
        >
          <div
            class="mantine-TextInput-root mantine-10rtrbx"
          >
            <label
              class="mantine-1w07r5r mantine-TextInput-label"
              for="mantine-exqmqpxs1"
              id="mantine-exqmqpxs1-label"
            >
              Email
              <span
                aria-hidden="true"
                class="mantine-1m203yh mantine-TextInput-required"
              >
                 *
              </span>
            </label>
            <div
              class="mantine-TextInput-wrapper mantine-12sbrde"
            >
              <input
                aria-invalid="false"
                class="mantine-TextInput-defaultVariant mantine-TextInput-input mantine-g773hu"
                id="mantine-exqmqpxs1"
                placeholder="Enter your Email"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="mantine-TextInput-root mantine-1x9761l"
          >
            <label
              class="mantine-1w07r5r mantine-TextInput-label"
              for="mantine-zwd3hgd2t"
              id="mantine-zwd3hgd2t-label"
            >
              Password
              <span
                aria-hidden="true"
                class="mantine-1m203yh mantine-TextInput-required"
              >
                 *
              </span>
            </label>
            <div
              class="mantine-TextInput-wrapper mantine-12sbrde"
            >
              <input
                aria-invalid="false"
                class="mantine-TextInput-defaultVariant mantine-TextInput-input mantine-15cu3qq"
                id="mantine-zwd3hgd2t"
                placeholder="Password"
                required=""
                type="password"
                value=""
              />
              <div
                class="mantine-1n9cu5e mantine-TextInput-rightSection"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  style="cursor: pointer;"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m10.79 12.912-1.614-1.615a3.5 3.5 0 0 1-4.474-4.474l-2.06-2.06C.938 6.278 0 8 0 8s3 5.5 8 5.5a7.029 7.029 0 0 0 2.79-.588zM5.21 3.088A7.028 7.028 0 0 1 8 2.5c5 0 8 5.5 8 5.5s-.939 1.721-2.641 3.238l-2.062-2.062a3.5 3.5 0 0 0-4.474-4.474L5.21 3.089z"
                  />
                  <path
                    d="M5.525 7.646a2.5 2.5 0 0 0 2.829 2.829l-2.83-2.829zm4.95.708-2.829-2.83a2.5 2.5 0 0 1 2.829 2.829zm3.171 6-12-12 .708-.708 12 12-.708.708z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="mantine-Text-root mantine-y2od05"
          />
          <button
            class="mantine-Button-filled mantine-Button-root mantine-1fhoz4r"
            type="submit"
          >
            <div
              class="mantine-3xbgk5 mantine-Button-inner"
            >
              <span
                class="mantine-qo1k2 mantine-Button-label"
              >
                Login
              </span>
            </div>
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
